internal/v2/tls_config_store/tls_config_store.go

```diff
@@ -0,0 +1,114 @@
+// Static implementation of TLS Configuration Store (no calls to S2Av2, Remote Signer Library, Certificate Verifier)
+package tls_config_store
+
+import (
+	"log"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"time"
+
+	_ "embed"
+)
+
+var (
+	//go:embed example_cert_key/client_cert.pem
+	clientCert []byte
+	//go:embed example_cert_key/server_cert.pem
+	serverCert []byte
+	//go:embed example_cert_key/client_key.pem
+	clientKey []byte
+	//go:embed example_cert_key/server_key.pem
+	serverKey []byte
+)
+
+func VerifyPeerCertificateFunc(instanceName string, pool *x509.CertPool) func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+	return func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+		if len(rawCerts) == 0 {
+			return fmt.Errorf("no certificate to verify")
+		}
+		cert, err := x509.ParseCertificate(rawCerts[0])
+		if err != nil {
+			return fmt.Errorf("x509.ParseCertificate(rawCerts[0]) returned error: %v", err)
+		}
+
+		opts := x509.VerifyOptions{
+			CurrentTime: time.Now(),
+			Roots:       pool,
+		}
+
+		if _, err = cert.Verify(opts); err != nil {
+			return err
+		}
+
+		if cert.Subject.CommonName != instanceName {
+			return fmt.Errorf("certificate had Common Name %q, expected %q", cert.Subject.CommonName, instanceName)
+		}
+		return nil
+	}
+}
+
+
+func GetTlsConfigurationForClient(id string) tls.Config {
+	// Static implementation. TODO : Call S2Av2 for these values.
+	min_version := uint16(tls.VersionTLS13)
+	max_version := uint16(tls.VersionTLS13)
+	var cipher_suites []uint16
+	var curve_preferences []tls.CurveID
+
+	// Static implementation. TODO : Call remote signer library for Private Key.
+	cert, err := tls.X509KeyPair(clientCert, clientKey)
+	if err != nil {
+		log.Fatalf("Failed to get client cert")
+	}
+
+	rootCertPool := x509.NewCertPool()
+	rootCertPool.AppendCertsFromPEM(serverCert)
+
+	// Create mTLS credentials for client.
+	config := tls.Config {
```

Comment:
2 micro-nits:

* Should we directly return rather than storing into the `config` variable?

* Should we return a pointer instead of the struct itself? In this case, we would need to write `return &tls.Config{...}`

Reply:
Thanks for pointing this out! I didn't know this was a feature but I found this(https://tip.golang.org/src/cmd/compile/internal/escape/escape.go), and if I interpreted it correctly, Go performs escape analysis, so in this case, if we return &tls.Config{...}, the Go compiler will allocate memory for it on the heap, whereas in other languages, returning a pointer to an object on the local stack would cause a memory error.